### PR TITLE
Change cache directory structure to include `pytest`

### PR DESCRIPTION
**Pull request by alanbato**
Hi all. This should deal with the issue discussed in #3138. I've basically done the following:

- Change the `v` folder to `pytest_values`
- Change the `cache/lastfailed` directory/file to just `lastfailed` inside the `pytest_values` directory
- Change the tests accordingly.

----
Thanks for submitting a PR, your contribution is really appreciated!

Here's a quick checklist that should be present in PRs:

- [x] Add a new news fragment into the changelog folder
  * name it `$issue_id.$type` for example (588.bugfix)
  * if you don't have an issue_id change it to the pr id after creating the pr
  * ensure type is one of `removal`, `feature`, `bugfix`, `vendor`, `doc` or `trivial`
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."
- [x] Target: for `bugfix`, `vendor`, `doc` or `trivial` fixes, target `master`; for removals or features target `features`;
- [x] Make sure to include reasonable tests for your change if necessary

Unless your change is a trivial or a documentation fix (e.g.,  a typo or reword of a small section) please:

- [x] Add yourself to `AUTHORS`, in alphabetical order;

### CODE REVIEW

**Comment by nicoddemus** [comment]
Hi @alanbato thanks for the PR!

There are a couple of issues in my POV though:

1. Currently the root directory is `.cache`; I believe this is the heart of #3138, that there's no indication that this directory has anything to do with pytest; I believe we should try to add the `pytest` name in there, perhaps just renaming it to `.pytest_cache`.
2. The internal structure should not really change. The keys defined by the users of the "cache" API should always include some kind of application name (`cache/lastfailed`, `myapp/some_info`, etc) to avoid naming clashes.

In summary I think we should just change the root directory from `.cache` to `.pytest_cache`. 😁

**Comment by alanbato** [comment]
Oh, I totally misunderstood what was required then 😅 
I'll revert back my changes and work on your idea right now :)

**Comment by nicoddemus** [comment]
@alanbato the description could be clearer actually, my apologies. 

Thanks for working on this. 👍

**Comment by alanbato** [comment]
My pleasure. Hopefully I can work on some other minor issues and bugs and go up from there :)

By the way @nicoddemus, just to be sure, this should be regarded as a 'feature', right? Or is another label more appropiate? (For the changelog and such)

**Comment by nicoddemus** [comment]
> My pleasure. Hopefully I can work on some other minor issues and bugs and go up from there :)

That's awesome! Please don't hesitate to ask for help if you run into any roadblocks.

> By the way @nicoddemus, just to be sure, this should be regarded as a 'feature', right? Or is another label more appropiate? (For the changelog and such)

Yeah it is a "feature" in the sense that it is a new behavior which is not an actual bug-fix. 👍

**Comment by nicoddemus** [comment]
Thanks @alanbato, LGTM now.

We should expect some complaints because people will need to update their `.gitignore` files, but other than that I don't expect too much disruption.

@RonnyPfannschmidt any thoughts?

**Comment by The-Compiler** [comment]
What about old `.cache/v/cache/lastfailed` files (and potentially plugins putting their own stuff there)? Shouldn't we ideally move things over? That comes with quite a few corner cases though, and for `lastfailed` it's not critical (but it's still old files rotting) - however, for other information it might be?

**Comment by RonnyPfannschmidt** [comment]
@The-Compiler since its a "cache" i consider a migration tool unnecessary, however we might want to support the old folder of it exists

**Comment by nicoddemus** [comment]
> since its a "cache" i consider a migration tool unnecessary

I agree, critical information should not be stored there, and a migration tool seems unnecessary after all a simple `mv .cache .pytest_cache` is enough (we might want to document that). 😁 

> however we might want to support the old folder of it exists

My initial take on this looked hard to implement, but what if we just move `.cache` to `.pytest_cache` during the plugin initialization temporarily while in `3.4`? We can then remove this code in `3.5` or later when we expect most users to already have the cache folder under the new name.

**Comment by RonnyPfannschmidt** [comment]
well - the current state is mergeable from my pov, lets have the sorting of compatibility for afterwards, as its not critical

**Comment by nicoddemus** [comment]
Thanks again @alanbato for the contribution! 👍
